pkgs/by-name/in/inngest/package.nix

```
@@ -0,0 +1,114 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  stdenv,
+  pnpm_8,
+  nodejs_20,
+  writeScript,
+}:
+let
+  version = "1.11.10";
+  shortCommit = "7a88e1e";
+
+  src = fetchFromGitHub {
+    owner = "inngest";
+    repo = "inngest";
+    tag = "v${version}";
+    hash = "sha256-RAMr1p0HnEEyxqnLCFmK6XiGYZu2NelGhBQSrEYEuBg=";
+  };
+
+  inngest-ui = stdenv.mkDerivation (finalAttrs: {
+    inherit version src;
+    pname = "inngest-ui";
+
+    nativeBuildInputs = [
+      pnpm_8
+      pnpm_8.configHook
+      nodejs_20
+    ];
+
+    buildPhase = ''
+      cd ui/apps/dev-server-ui && pnpm build
+    '';
+
+    installPhase = ''
+      mkdir -p $out
+      cp -r dist $out
+      cp -r .next/routes-manifest.json $out/dist
+    '';
+
+    pnpmWorkspaces = [ "dev-server-ui" ];
+
+    pnpmDeps = pnpm_8.fetchDeps {
+      inherit (finalAttrs) pname pnpmWorkspaces;
+      inherit version src;
+      hash = "sha256-wBljTZUrq673QEJWakfm85ikVgxtBgimzWlQxxacOXM=";
+      sourceRoot = "${finalAttrs.src.name}/ui";
+      fetcherVersion = 1;
+    };
+    pnpmRoot = "ui";
+  });
+in
+buildGoModule {
+  inherit version src inngest-ui;
+  pname = "inngest";
+
+  vendorHash = null;
+
+  preBuild = ''
+    go generate ./...
+    cp -r ${inngest-ui}/dist/* ./pkg/devserver/static
+  '';
+
+  postInstall = ''
+    mv $out/bin/cmd $out/bin/inngest
+  '';
+
+  ldflags = [
+    "-s -w"
+    "-X github.com/inngest/inngest/pkg/inngest/version.Version=${version}"
+    "-X github.com/inngest/inngest/pkg/inngest/version.Hash=${shortCommit}"
+  ];
+
+  # The Inngest CI/CD uses GoReleaser to build the package, and the env `CGO_ENABLED` is set in the configuration file for GoReleaser
+  # https://github.com/inngest/inngest/blob/main/.goreleaser.yml#L9
+  env.CGO_ENABLED = 0;
+
+  subPackages = [ "cmd/" ];
+
+  passthru.updateScript = writeScript "update-inngest" ''
+    #!/usr/bin/env nix-shell
+    #!nix-shell -i nu -p nushell common-updater-scripts nix-update
```

Non-blocking suggestion: Add `-I` flag so that it uses the nixpkgs of the worktree, and not whatever is on your system.
```suggestion
    #!/usr/bin/env nix-shell
    #! nix-shell -I.
    #! nix-shell -i nu 
    #! nix-shell -p nushell common-updater-scripts nix-update
```